src/Java.Interop.Tools.JavaTypeSystem/JavaModels/JavaFieldModel.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Java.Interop.Tools.JavaTypeSystem.Models
+{
+	public class JavaFieldModel : JavaMemberModel
+	{
+		public string Type { get; }
+		public string TypeGeneric { get; }
+		public string? Value { get; }
+		public bool IsTransient { get; }
+		public bool IsVolatile { get; }
+		public bool IsNotNull { get; }
+
+		public JavaTypeReference? TypeReference { get; private set; }
+
+		public JavaFieldModel (string name, string visibility, string type, string typeGeneric, string? value, bool isStatic, JavaTypeModel parent, bool isFinal, string deprecated, string jniSignature, bool isTransient, bool isVolatile, bool isNotNull)
+			: base (name, isStatic, isFinal, visibility, parent, deprecated, jniSignature)
+		{
+			Type = type;
+			TypeGeneric = typeGeneric;
+			Value = value;
+			IsTransient = isTransient;
+			IsVolatile = isVolatile;
+			IsNotNull = isNotNull;
+		}
+
+		public override void Resolve (JavaTypeCollection types, List<JavaUnresolvableModel> unresolvables)
+		{
+			if (Name.Contains ('$')) {
```

Comment:
I don't understand this: `$` is a valid Java identifier character:

```java
public class ContainsDollar {
    public static int yay$Dollars;
}
```

It's *weird*, certainly, but it's *valid*.  Why does this cause the field to be unresolvable?

Reply:
Compatibility with existing code: https://github.com/xamarin/java.interop/commit/835aaf773f9f7e21df51f4f64c99945f04dd4e0f.